### ISSUE TITLE
use -b to output a json string with base64 encoded binary blobs

### ISSUE THREAD

**Issue by kapouer** [crussell42/exiftool-node-stayopen#2]
so you can parse binary output even with JSON.parse ! (i noticed you already use json output)

**Comment by crussell42** [comment]
Yes I can parse but often have to remove special characters before doing so.
Depends on the program used to insert the tags.
If you can be more specific about the problem I may be able to help.

**Comment by kapouer** [comment]
I'm not reporting bugs, here. More like ideas for enhancements.

>  but often have to remove special characters before doing so.

Do you mean it's not UTF-8 output ?

**Comment by crussell42** [comment]
All good. I have not spent much time on this lately so am not up on the
exact reasons.
I use this package in a production system and know that I had to scrub some
of the
returned json (could be utf8 issue but I honestly cant remember...would
have to dig into the program that uses the library to refresh my memory).
Feel free to give the lib a try and make what ever branches or improvements
you want.
I will help if I can spare the time.
I will try and update the wiki with a real world example of the
promise/result usage.

**Comment by kapouer** [comment]
That's cool. I'm using exiftool for https://github.com/kapouer/url-inspector and am investigating wether i can keep exiftool as a daemon and feed it with 50kb of file head then 100kb...and so on until it finds proper tags, to avoid downloading too much data. The `-fast` option is nice but not working on all file types. I'll tell you if i come up with something interesting.

**Comment by kapouer** [comment]
In https://github.com/kapouer/node-streat
i'm implementing those ideas.

It's probably a bit leaky though.
